src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/SG01.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.PerformingOrganisationNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RequesterNameAndAddress;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ServiceProvider;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class SG01 extends SegmentGroup {
+    public static final String KEY = "S01";
+
+    // NAD+SLA?
+    @Getter(lazy = true)
+    private final Optional<PerformingOrganisationNameAndAddress> performingOrganisationNameAndAddress =
+        extractOptionalSegment(PerformingOrganisationNameAndAddress.KEY_QUALIFIER)
+            .map(PerformingOrganisationNameAndAddress::fromString);
+
+    // NAD+PO?
+    @Getter(lazy = true)
+    private final Optional<RequesterNameAndAddress> requesterNameAndAddress =
+        extractOptionalSegment(RequesterNameAndAddress.KEY_QUALIFIER)
+            .map(RequesterNameAndAddress::fromString);
+
+    // SPR
+    @Getter(lazy = true)
+    private final ServiceProvider serviceProvider = ServiceProvider.fromString(extractSegment(ServiceProvider.KEY));
+
+    public static List<SG01> createMultiple(@NonNull final List<String> sg01Segments) {
+        return splitMultipleSegmentGroups(sg01Segments, KEY).stream()
+            .map(SG01::new)
+            .collect(toList());
+    }
```

Comment:
The approach to derive create multiple instances of a segment group is common across segment groups. This method expects that the list of segments passed to it contains a headered sequence of the desired segments, including their children, but nothing else afterwards. If the first segment is not the expected header, it assumes the calling code did not actually find any of the segment group.